api/apps/articles/model.coffee

```diff
@@ -295,9 +296,15 @@ generateSlugs = (article, author, cb) ->
     article.slugs = (article.slugs or []).concat slug
     cb(null, article)
 
-generateKeywords = (article, accessToken, cb) ->
+generateKeywords = (article, accessToken, input, cb) ->
   keywords = []
   callbacks = []
+  if (input.primary_featured_artist_ids is not article.primary_featured_artist_ids or
+      input.featured_artist_ids is not article.featured_artist_ids or
+      input.fair_id is not article.fair_id or
+      input.partner_ids is not article.partner_ids or
+      input.tags is not article.tags)
+    return cb(null, article)
```

Comment:
Kinda nitpicking here, but since this is now happening before merging input & article data can we move it above [this comment]((https://github.com/kanaabe/positron/blob/daily_email/api/apps/articles/model.coffee#L283)) and move the function call [before the merge step](https://github.com/kanaabe/positron/blob/daily_email/api/apps/articles/model.coffee#L251). Also might be time to think about refactoring all of this stuff into it's own `onsave.coffee` module that just exports the functions used in `@save` like `validate -> generateKeywords -> mergeArticleAndAuthor -> onPublish/generateSlugs/sanitizeAndSave`